packages/x-date-pickers-pro/src/DateRangeCalendar/DateRangeCalendar.tsx

```diff
@@ -108,13 +112,45 @@ function useDateRangeCalendarDefaultizedProps<TDate extends PickerValidDate>(
   props: DateRangeCalendarProps<TDate>,
   name: string,
 ): DateRangeCalendarDefaultizedProps<TDate> {
+  const {
+    shouldDisableDate,
+    shouldDisableMonth,
+    shouldDisableYear,
+    disablePast,
+    disableFuture,
+    timezone,
+    value
+  } = props;
   const utils = useUtils<TDate>();
   const defaultDates = useDefaultDates<TDate>();
   const defaultReduceAnimations = useDefaultReduceAnimations();
   const themeProps = useThemeProps({
     props,
     name,
   });
+  const isDateDisabled = useIsDateDisabled({
+    shouldDisableDate,
+    shouldDisableMonth,
+    shouldDisableYear,
+    minDate: themeProps.minDate || defaultDates.minDate,
+    maxDate: themeProps.maxDate || defaultDates.maxDate,
+    disablePast,
+    disableFuture,
+    timezone,
+  });
+
+  const [start, end] = value;
+  let maxDateNew = themeProps.maxDate ?? defaultDates.maxDate;
+  let minDateNew = themeProps.minDate ?? defaultDates.minDate;
+
+  if(start) {
```

Comment:
This logic is intended to be removed from here, since it's a hook dedicated to default props.
My first thought is to create something like `useEnrichedDateRangeCalendarProps` similar to what exists for Fields, so we can have a place to enrich and override dateRangeCalendar default props. Suggestions are very welcome ! 🙂

Reply:
We have a complication—`rangePosition` prop can be controlled and there is also the `defaultRangePosition` that could be set to `end`.
This means that we will need to support non-contigous range validation in other direction (changing `minDate`). 🙈

Reply:
This logic also has to apply to the `Field` components, doesn't it? 🤔 
In this case, going for `useEnrichedDateRangeCalendarProps` might not be the best approach. 🙈